### PR TITLE
Getting started with the user guide.

### PR DESCRIPTION
**Pull request by novemberkilo**
Screenshot 2022-11-28 at 3 43 28 pm

cc/ @christinerose

### CODE REVIEW

File: web-ui/view/documentation.ml [hunk 1]
```diff
@@ -103,3 +106,177 @@ let getting_started =
...
+          div
+            ~a:[ a_class [ "prose col-span-4" ] ]
+            [
+              h1 [ txt "Documentation" ];
+              p
+                [
+                  txt
+                    "This is a Work In Progress. Thank you for your patience \
+                     as we update it with detailed documentation covering each \
+                     of the pages of ocaml-ci.";
```
Comment (maiste) [inline web-ui/view/documentation.ml:132]: Should we be clear that the `ocaml-ci` project globally is also a WIP, not just the documentation?

File: web-ui/view/documentation.ml [hunk 2]
```diff
@@ -103,3 +106,177 @@ let getting_started =
...
+              p
+                [
+                  txt "The generated Dockerfile first adds all the ";
+                  code [ txt "*.opam" ];
+                  txt
+                    " files found in the project, then uses opam to install \
+                     all the dependencies, and finally adds the rest of the \
+                     source files. This means that rebuilds are often very \
+                     fast, because Docker will reuse the previously cached \
+                     build step as long as the opam files don’t change.";
```
Comment (maiste) [inline web-ui/view/documentation.ml:206]: Can't we add a sentence saying that it tries to install all the opam files at the same time and pin the version to the local version? This way, we ensure to discard all exotic configurations that require extra steps.

**Review by maiste — APPROVED** [review]
Few questions but LGTM :+1:

**Review by benmandrew — APPROVED** [review]
I think the backticks look a little strange in the web-UI for showing files, as I'd only expect to see backticks in raw MD files. We could perhaps use something along the lines of the Github style?
Just personal preference though, if you disagree feel free to continue.

File: web-ui/view/documentation.ml [hunk 3]
```diff
@@ -103,3 +106,177 @@ let getting_started =
...
+                  txt
+                    "OCaml-CI is a continuous integration (CI) tool for OCaml \
+                     projects. This development practice encourages developers \
+                     to regularly integrate code into a shared repository. \
+                     Commits are verified by an automated build, allowing \
+                     teams to detect and fix problems early.";
+                ];
+              p
+                [
+                  txt "OCaml-CI uses metadata from the project’s opam and ";
```
[the remark this answers is not in the document]
Comment (christinerose) [reply web-ui/view/documentation.ml:146]: Is there a way to show monospace in the web-UI?

File: web-ui/view/documentation.ml (hunk 1 above)
Comment (christinerose) [inline web-ui/view/documentation.ml:132]:
```suggestion
                     of the pages of OCaml-CI.";
```

**Review by christinerose — CHANGES_REQUESTED** [review]

File: web-ui/view/documentation.ml (hunk 3 above)
[the remark this answers is not in the document]
Comment (christinerose) [reply web-ui/view/documentation.ml:146]: Oh! Does this go through `odoc`? I have a note that monospace in .ml and .mli files are surrounded by [ ]

File: web-ui/view/documentation.ml (hunk 2 above)
Comment (novemberkilo) [reply web-ui/view/documentation.ml:206]: This seems important but I don't understand this enough to write this simply and correctly. Can you please suggest an edit?

File: web-ui/view/documentation.ml (hunk 3 above)
[the remark this answers is not in the document]
Comment (novemberkilo) [reply web-ui/view/documentation.ml]: Okay I have ripped some code off some part of the Internet to get rid of the backticks.

Screenshot 2022-11-29 at 8 16 51 am

File: web-ui/view/documentation.ml (hunk 1 above)
Comment (novemberkilo) [reply web-ui/view/documentation.ml]: This appears in the Getting Started page so I think we can skip it here.

File: web-ui/view/documentation.ml (hunk 3 above)
[the remark this answers is not in the document]
Comment (christinerose) [reply web-ui/view/documentation.ml]: Thank you!

File: tailwind.config.js [hunk 4]
```diff
@@ -10,6 +10,30 @@ module.exports = {
...
           500: "#F79009",
         }
+      },
+      typography (theme) {
+        return {
+          DEFAULT: {
+            css: {
+              'code::before': {
+                content: 'none', // don’t generate the pseudo-element
+//                content: '""', // this is an alternative: generate pseudo element using an empty string
```
Comment (tmcgilchrist) [inline tailwind.config.js:20]:
```suggestion
```

File: web-ui/view/documentation.ml [hunk 5]
```diff
@@ -103,3 +108,179 @@ let getting_started =
...
+          div
+            ~a:[ a_class [ "prose col-span-4" ] ]
+            [
+              h1 [ txt "Documentation" ];
+              p
+                [
+                  txt
+                    "This is a Work In Progress. Thank you for your patience \
+                     as we update it with detailed documentation covering each \
+                     of the pages of Ocaml-ci.";
```
Comment (tmcgilchrist) [inline web-ui/view/documentation.ml:134]:
```suggestion
                     of the pages of OCaml-ci.";
```

File: web-ui/view/documentation.ml (hunk 2 above)
Comment (tmcgilchrist) [reply web-ui/view/documentation.ml:210]: An example would be clearer here.

Given a project with the following opam files
```
bondi.opam
bondi-lwt.opam
bondi-async.opam
```
OCaml-CI will run an opam solve for all of the dependencies in all the opam files together (implied by the opam install but perhaps not clear enough). So if you depend on OCaml 5 in `bondi-lwt.opam` then a build will only happen on that version of OCaml, even if the other two packages could run on earlier versions of OCaml.

See https://github.com/ocurrent/ocaml-ci/issues/470 for the issue suggesting an alternate build strategy.

Comment (maiste) [reply web-ui/view/documentation.ml:210]: > OCaml-CI will run an opam solve for all of the dependencies in all the opam files together
Not with the right wording but that's what I meant ^^

Comment (christinerose) [reply web-ui/view/documentation.ml:210]: How about: 

OCaml-CI will run an opam solve for all dependencies in every opam file simultaneously.

**Review by christinerose — APPROVED** [review]

**Comment by MisterDA** [comment]
> I think the backticks look a little strange in the web-UI for showing files, as I'd only expect to see backticks in raw MD files. We could perhaps use something along the lines of the Github style? Just personal preference though, if you disagree feel free to continue.

100% agreed. Possibly related to https://github.com/ocaml/ocaml.org/pull/625?

**Pull request merged**